presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/InMemoryHashAggregationBuilder.java

```diff
@@ -348,14 +395,38 @@ public Type getType()
 
         public void processPage(GroupByIdBlock groupIds, Page page)
         {
-            if (step.isInputRaw()) {
+            if (!this.orderByChannels.isEmpty()) {
+                checkArgument(step == Step.SINGLE && pageIndex.isPresent(), "Order by over distributed aggregation is not supported");
```

Comment:
All keys in the same group will always end up on a same node, regardless of the step (this will happen in either `SINGLE` step or in `FINAL` step after partial aggregation). And it has nothing to do with single node/distributed execution.

I think that I understand what's for this checkstate, only error message is confusing/inadequate. Please change it to something like: 
```
checkState(step == Step.SINGLE && pageIndex.isPresent(), "ORDER BY in aggregations is supported only for single step aggregations");
```